### ISSUE TITLE
Alternative: make :visited main-frame only

### ISSUE THREAD

**Issue by RByers** [kyraseevers/Partitioning-visited-links-history#1]
One variation not discussed is to make :visited apply only in main frames and to main frame history entries. I.e.  double-key rather than triple-key. 

Do you know of legitimate use cases for :visited in iframes? Any idea if anyone would notice if that stopped working? Would that save much in terms of complexity, RAM etc?

I suppose this could always be a potential future signification based on metrics of the cost / benefit tradeoff right?

**Comment by arturjanc** [comment]
> Seems in its current state sandboxed pages and embeds would get there visited status removed after a reload.

I think removing state from sandboxed frames is more of a feature than a bug -- documents from opaque origins aren't supposed to maintain state (e.g. no cookies, `localStorage`, etc.). So it seems reasonable to uphold this for history as well and ignore visitedness from sandbox frames.

Re: limiting :visited to the main frame, I see how it could let us avoid some complexity. However, it would also reduce some of the utility of visitedness, e.g. in scenarios with same-origin iframes; I'm a little worried that it might limit functionality in a way we could avoid if we went with a triple-keyed design (but AFAIK we don't have any data to either support or disprove this).

Another benefit of triple-keying is that it aligns the partitioning design with what we've done for HTTP cache and network state, so there's some conceptual consistency in how we're separating various kinds of client-side state.

**Comment by NDevTK** [comment]
Yeah feature just wanted to ensure that frame origin stayed included.
So no leaking visitedness without allow-same-origin

It seems unlikely visited links can just be disabled for all embeds.
